app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/catalog/ui/view/FourthPageContent.kt

```diff
@@ -0,0 +1,196 @@
+
+    PageLayout(
+        modifier,
+        pageNumberOrdinal + 1,
+        catalogList.sizeOrZero()
+    ) {
+        FourthPageContent(
+            modifier
+                .padding(
+                    start = if (isFeatureHorizontal)
+                        dimensionResource(id = R.dimen.catalog_page_padding) else
+                        dimensionResource(id = R.dimen.zero_padding),
+                    top = if (isFeatureHorizontal)
+                        dimensionResource(id = R.dimen.catalog_margin_normal) else
+                        dimensionResource(id = R.dimen.zero_padding)
+                )
+                .verticalScroll(rememberScrollState()),
+            fourthPageConstraintSet,
+            catalogItem,
+            isFeatureHorizontal
+        )
+    }
+}
+
+private fun getConstraintSetForFourthPage(isFeatureHorizontal: Boolean) = ConstraintSet {
+    val firstImageRef = createRefFor(FOURTH_PAGE_FIRST_IMAGE_ID)
+    val secondImageRef = createRefFor(FOURTH_PAGE_SECOND_IMAGE_ID)
+    val firstTextRef = createRefFor(FOURTH_PAGE_FIRST_TEXT_ID)
+    val secondTextRef = createRefFor(FOURTH_PAGE_SECOND_TEXT_ID)
+    val thirdTextRef = createRefFor(FOURTH_PAGE_THIRD_TEXT_ID)
+    val horizontalGuideline = createGuidelineFromTop(0.5f)
+
+    constrain(firstTextRef) {
+        linkTo(start = parent.start, end = parent.end)
+        top.linkTo(firstImageRef.bottom)
+        if (isFeatureHorizontal) {
+            bottom.linkTo(horizontalGuideline)
+        }
+    }
+
+    constrain(secondTextRef) {
+        top.linkTo(if (isFeatureHorizontal) horizontalGuideline else firstTextRef.bottom)
+        start.linkTo(firstTextRef.start)
+    }
+
+    constrain(thirdTextRef) {
+        linkTo(start = firstTextRef.start, end = firstTextRef.end)
+        top.linkTo(secondTextRef.bottom)
+    }
+
+    createHorizontalChain(firstImageRef, secondImageRef, chainStyle = ChainStyle.Spread)
+}
+
+@Composable
+fun FourthPageContent(
+    modifier: Modifier,
+    constraintSet: ConstraintSet,
+    catalogItem: CatalogItem,
+    isFeatureHorizontal: Boolean
+) {
+    ConstraintLayout(
+        constraintSet = constraintSet,
+        modifier = modifier
+    ) {
+        RoundedImage(
+            modifier = Modifier
+                .padding(top = dimensionResource(id = R.dimen.catalog_top_margin))
+                .layoutId(FOURTH_PAGE_FIRST_IMAGE_ID)
+                .requiredWidth(
+                    if (isFeatureHorizontal)
+                        dimensionResource(id = R.dimen.catalog_max_image_height_dual_landscape) else
+                        dimensionResource(id = R.dimen.catalog_min_image_height)
+                )
+                .requiredHeight(
+                    if (isFeatureHorizontal)
+                        dimensionResource(id = R.dimen.catalog_max_image_height) else
+                        dimensionResource(id = R.dimen.catalog_min_image_height)
+                ),
```

Comment:
I will test the app on different devices after reviewing the code, but I would just say to be careful with `requiredWidth` and `requiredHeight` as these sizes may not look good on devices other than Surface Duo - instead, you could use the [heightIn](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/package-summary#(androidx.compose.ui.Modifier).heightIn(androidx.compose.ui.unit.Dp,androidx.compose.ui.unit.Dp)), [widthIn](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/package-summary#(androidx.compose.ui.Modifier).widthIn(androidx.compose.ui.unit.Dp,androidx.compose.ui.unit.Dp)), or [sizeIn ](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/package-summary#(androidx.compose.ui.Modifier).sizeIn(androidx.compose.ui.unit.Dp,androidx.compose.ui.unit.Dp,androidx.compose.ui.unit.Dp,androidx.compose.ui.unit.Dp))attributes (especially if the size ranges are already defined in the view layouts, then you can just copy those dimensions since Bianca already tested those out)

Reply:
I've tested the app with Surface Duo 2, Samsung Galaxy Z Fold3, Pixel 4&5, of course you can test the app as well. Initially I was trying to  use the heighIn and widthIn methods but for some reason that didn't work for my case, but requiredHeight and requiredWidth solved the problem.